libnavigation-ui/src/main/java/com/mapbox/navigation/ui/NavigationView.java

```diff
@@ -322,17 +322,25 @@ public void updateWayNameView(@NonNull String wayName) {
     wayNameView.updateWayNameText(wayName);
   }
 
-  /**
-   * Updates the visibility of the way name view that is show below
-   * the navigation icon.
-   * <p>
-   * If you'd like to use this method without being overridden by the default visibility values
-   * values we provide, please disabled auto-query with
-   * {@link NavigationMapboxMap#updateWaynameQueryMap(boolean)}.
-   *
-   * @param isVisible true to show, false to hide
-   */
   @Override
+  public void setWayNameActive(boolean isActive) {
+    if (navigationMap != null) {
+      navigationMap.updateWaynameQueryMap(isActive);
+    }
+  }
+
+  @Override
+  public void setWayNameVisibility(boolean isVisible) {
+    wayNameView.updateVisibility(isVisible);
+  }
+
+  @Override
+  public String retrieveWayNameText() {
+    return wayNameView.retrieveWayNameText();
+  }
+
+  @Override
+  @Deprecated
```

Comment:
What if we can just remove anything(without deprecate) until we are in beta? cc @LukasPaczos @Guardiola31337

Reply:
That's a great question but are we technically breaking the APIs / is `libnavigation-ui` going to be a major release? 🤔

Reply:
We break, but it's not a big problem, we use it internally, and it's easy to fix in UI module. I'm curious that we might have few deprecation, that would be great to restrict before stable version 😃